node/src/sub_lib/node_addr.rs

```diff
@@ -12,7 +12,7 @@ use std::str::FromStr;
 
 #[derive(PartialEq, Eq, Hash, Deserialize, Serialize)]
 pub struct NodeAddr {
-    ip_addr: IpAddr,
+    pub(crate) ip_addr: IpAddr,
```

Comment:
This is not a way how problems should be solved. Reconsider if there isn't a different way to get this information without making the field public. 

How? Make it private again and try compiling the code. Then look at the popped errors and see if you can get around it with a better design. 

More importantly, I believe that NodeAddr has already a method by which you can ask about the ip_addr, which is exactly what you should have done in the first place.